### PR TITLE
Add support for aiohttp client and server

### PR DESCRIPTION
**Pull request by beniwohli**
## What does this pull request do?

This PR adds integration for the aiohttp web framework, as well as the aiohttp
HTTP client library.

## Why is it important?

~Adding support for more asyncio frameworks and libraries is part of the roadmap.~ BECAUSE ASYNC ALL THE THINGS! (edited for excitement by @basepi)

### CODE REVIEW

File: elasticapm/contrib/aiohttp/middleware.py [hunk 1]
```diff
@@ -31,11 +31,16 @@
...
 import elasticapm
-from elasticapm.conf import constants
 from elasticapm.contrib.aiohttp.utils import get_data_from_request, get_data_from_response
 from elasticapm.utils.disttracing import TraceParent
 
 
+class AioHttpTraceParent(TraceParent):
+    @classmethod
+    def merge_duplicate_headers(cls, headers, key):
+        return ",".join(headers.getall(key))
```
Comment (basepi) [inline elasticapm/contrib/aiohttp/middleware.py]: Good catch. And validates your decision to make the dummy method as well! :)

File: elasticapm/instrumentation/packages/asyncio/aiohttp_client.py [hunk 2]
```diff
@@ -72,6 +72,14 @@ def mutate_unsampled_call_args(self, module, method, wrapped, instance, args, kw
...
         kwargs["headers"] = headers
         return args, kwargs
+
+    def _set_disttracing_headers(self, headers, trace_parent, transaction):
+        trace_parent_str = trace_parent.to_string()
+        headers[constants.TRACEPARENT_HEADER_NAME] = trace_parent_str
+        if transaction.tracer.config.use_elastic_traceparent_header:
+            headers[constants.TRACEPARENT_LEGACY_HEADER_NAME] = trace_parent_str
+        if trace_parent.tracestate:
+            headers[constants.TRACESTATE_HEADER_NAME] = trace_parent.tracestate
```
Comment (basepi) [inline elasticapm/instrumentation/packages/asyncio/aiohttp_client.py]: We'll probably want to keep an eye on these types of functions as we go along. Has potential to become a maintenance problem if this behavior changes over time. Though I guess searching for the `constants.TRACEPARENT` will probably be sufficient for any number of uses.

File: elasticapm/contrib/aiohttp/middleware.py (hunk 1 above)
Comment (beniwohli) [reply elasticapm/contrib/aiohttp/middleware.py]: To be honest, looking at aiohttp led me to add that dummy method in the first place 😅 no crystal balls involved I'm afraid

File: elasticapm/contrib/aiohttp/middleware.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        elasticapm_client = app.get(CLIENT_KEY)
+        if elasticapm_client:
+            request[CLIENT_KEY] = elasticapm_client
+            trace_parent = AioHttpTraceParent.from_headers(request.headers)
+            elasticapm_client.begin_transaction("request", trace_parent=trace_parent)
+            resource = request.match_info.route.resource
+            if resource:
+                name = "{} {}".format(request.method, resource.canonical)
+            else:
+                name = "unkown route"
```
Comment (basepi) [inline elasticapm/contrib/aiohttp/middleware.py]:
```suggestion
                name = "unknown route"
```

File: docs/aiohttp-server.asciidoc [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+}
+apm = ElasticAPM(app)
+----
+
+[float]
+[[aiohttp-server-usage]]
+==== Usage
+
+Once you have configured the agent,
+it will automatically track transactions and capture uncaught exceptions within Flask.
```
Comment (basepi) [inline docs/aiohttp-server.asciidoc]: > Flask

copy pasta?

File: docs/aiohttp-server.asciidoc [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+If you've followed the instructions above, the agent has already installed our middleware.
+This will measure response times, as well as detailed performance data for all supported technologies.
+
+NOTE: due to the fact that `asyncio` drivers are usually separate from their synchronous counterparts,
+specific instrumentation is needed for all drivers.
+The support for asynchronous drivers is currently quite limited.
+
+[float]
+[[aiohttp-server-ignoring-specific-views]]
+===== Ignoring specific routes
```
Comment (basepi) [inline docs/aiohttp-server.asciidoc]: We get this behavior for free, right? This isn't handled in your integration, but rather in the coordinating code, correct? Just taking note for my tornado stuff.

Comment (beniwohli) [reply docs/aiohttp-server.asciidoc]: Excellent question, had to look it up myself :D. It happens in `Tracer.end_transaction()`, so is framework independent.

https://github.com/elastic/apm-agent-python/blob/ac3b6ffb5893f1b989507eaa671ff8fbcaa53685/elasticapm/traces.py#L525-L549

**Pull request merged**